src/Dialect/ONNX/Rewrite.td

```diff
@@ -576,12 +576,6 @@ def SwapTransposeConcatPattern: Pat<
 // Canonicalization for ONNXReshapeOp
 //===----------------------------------------------------------------------===//
 
-def FuseReshapePattern:  Pat<
```

Comment:
could we keep this when az2 is false?

Reply:
I guess you mean `az2 == 1`. I restores this and checked `az1` and `az2`. I am not sure how it is when both `az1` and `az2` are 1 because I don't understand well a tensor of a mix of zero and non-zero dimensions. So only support `az1=0` and `az2=1`.

Reply:
sorry, I got mixed up

I think you can drop the `IntegerAttrIsOf<0> $az1)` requirement because when $az2==1 then $s2 describes the output shape completely regardless of $s1, if I understand correctly

if a tensor mixes zero and non-zero dimensions then the tensor is just empty (if any dimension is zero) and the spec says that then there can be no -1 (spec: "If the attribute 'allowzero' is set, it is invalid for the specified shape to contain both a zero value and -1, as the value of the dimension corresponding to -1 cannot be determined uniquely.")

Reply:
OK, so I removed the `az1` testing. Thanks!